packages/starlight/utils/path.ts

```diff
@@ -50,3 +50,14 @@ export function ensureHtmlExtension(path: string) {
 	}
 	return ensureLeadingSlash(path);
 }
+
+export function splitQueryAndFragment(url: string) {
```

Comment:
We should document this function probably like the other functions in this file:

```suggestion
/** Parse a URL into pathname, search params, and hash components. */
export function splitQueryAndFragment(url: string) {
```

Reply:
Will adjust once other feedback is resolved and final names determined.

Reply:
Eliminated `splitQueryAndFragment` util function as its no longer necessary given use of `URL` constructor.